app/controllers/api/v2/accredited_provider_training_providers_controller.rb

```diff
@@ -6,7 +6,21 @@ class AccreditedProviderTrainingProvidersController < API::V2::ApplicationContro
 
       def index
         authorize @provider, :can_list_training_providers?
-        providers = @provider.training_providers
+        if params[:filter]
+          course_ids = training_providers.map do |tp|
+            tp.courses.pluck(:id)
+          end.flatten
+
+          course_scope = Course.where(id: course_ids)
+
+          eligible_training_provider_ids = CourseSearchService
+                                             .call(filter: params[:filter], course_scope: course_scope)
+                                             .pluck(:provider_id)
+
+          providers = @training_providers.select { |provider| eligible_training_provider_ids.include?(provider.id) }
```

Comment:
is this a better option?
```
providers = @training_providers.where(id: eligible_training_provider_ids)
```

Reply:
Not so sure about this one? By using 'select' we save another db call ...